include/yaramod/types/token_stream.h

```diff
@@ -136,6 +142,7 @@ class TokenStream
 private:
 	std::list<Token> _tokens; ///< All tokens off the rule
 	bool formatted = false; ///< The flag is set once autoformat has been called
+	char new_line_char = '\n'; ///< The character used for line endings: usually '\n' on Unix or '\r' on Windows
```

Comment:
Shouldn't it be `\r\n` on Windows instead of just `\r`?

Reply:
So we will probably have to parse `\r\n` as a single character, not two different new line characters. Thank you for pointing this out.

Reply:
I fixed the problem. We insert LF or CR or CRLF depending on what is used in the file, more precisely which of the three line endings `\n`, `\r` or `\r\n` has been used last time.

Reply:
Great :+1: BTW, if you want to simplify the implementation a bit, I think you can safely ignore `\r` because it is no longer used (it was used in older macOS versions) and consider just `\n` (Linux, macOS) and `\r\n` (Windows). What do you think @metthal?

Reply:
I asked @metthal and he agrees with you so I will definitely simplify that: We will ignore `\r` because it is no longer used.